sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -0,0 +1,160 @@
+
+        /// <summary>
+        /// Decoded header for the attestation token. See https://tools.ietf.org/html/rfc7515 for more details.
+        /// </summary>
+        public ReadOnlyMemory<byte> TokenHeaderBytes { get; }
+
+        /// <summary>
+        /// Decoded body for the attestation token. See https://tools.ietf.org/html/rfc7515 for more details.
+        /// </summary>
+        public ReadOnlyMemory<byte> TokenBodyBytes { get; }
+
+        /// <summary>
+        /// Decoded signature for the attestation token. See https://tools.ietf.org/html/rfc7515 for more details.
+        /// </summary>
+        public ReadOnlyMemory<byte> TokenSignatureBytes { get; }
+
+        /// <summary>
+        /// Returns the standard properties in the JSON Web Token header. See https://tools.ietf.org/html/rfc7515 for more details.
+        /// </summary>
+        private JsonWebTokenHeader Header { get => JsonSerializer.Deserialize<JsonWebTokenHeader>(TokenHeaderBytes.ToArray()); }
+
+        /// <summary>
+        /// Returns the standard properties in the JSON Web Token header. See https://tools.ietf.org/html/rfc7515 for more details.
+        /// </summary>
+        private JsonWebTokenBody Payload { get => JsonSerializer.Deserialize<JsonWebTokenBody>(TokenBodyBytes.ToArray()); }
+
+        /// <summary>
+        /// Expiration time for the token.
+        /// </summary>
+        public DateTimeOffset ExpirationTime { get => DateTimeOffset.FromUnixTimeSeconds(Payload.ExpirationTime); }
+
+        /// <summary>
+        /// Time before which this token is not valid.
+        /// </summary>
+        public DateTimeOffset NotBeforeTime { get => DateTimeOffset.FromUnixTimeSeconds(Payload.NotBeforeTime); }
+
+        /// <summary>
+        /// Time at which this token was issued.
+        /// </summary>
+        public DateTimeOffset IssuedAtTime { get => DateTimeOffset.FromUnixTimeSeconds(Payload.IssuedAtTime); }
+
+
+        /// <summary>
+        /// Represents the body of the token encoded as a string.
+        /// </summary>
+        public string TokenBody {
+            get => Encoding.UTF8.GetString(TokenBodyBytes.ToArray());
+        }
+
+        /// <summary>
+        /// Represents the body of the token encoded as a string.
+        /// </summary>
+        public string TokenHeader
+        {
+            get => Encoding.UTF8.GetString(TokenHeaderBytes.ToArray());
+        }
+
+        /// <summary>
+        /// Validate a JSON Web Token returned by the MAA.
+        /// </summary>
+        /// <param name="attestationSigningCertificates">Signing Certificates used to validate the token.</param>
+        /// <param name="validationCallback">User provided callback which allows the customer to validate the token.</param>
+        /// <returns></returns>
+        public virtual bool ValidateToken(IReadOnlyList<AttestationSigner> attestationSigningCertificates, Func<AttestationToken, AttestationSigner, bool> validationCallback = default)
```

Comment:
Generally, input collections would be typed as IEnumerable. Is there a reason this is an IReadOnlyList?

Reply:
It's an IReadOnlyList because Krzystof said it should be a IReadOnlyList during the API review. I don't honestly care one way or another.